src/uct/ib/rc/accel/rc_mlx5_devx.c

```diff
@@ -140,12 +140,17 @@ uct_rc_mlx5_devx_init_rx_tm(uct_rc_mlx5_iface_common_t *iface,
     return status;
 }
 
-int uct_rc_mlx5_is_devx_srq_supported(const uct_ib_mlx5_md_t *md, int dc)
+int uct_rc_mlx5_is_devx_srq_supported(const uct_rc_mlx5_iface_common_t *iface,
+                                      int dc)
 {
+    uct_ib_mlx5_md_t *md   = ucs_derived_of(iface->super.super.super.md,
+                                            uct_ib_mlx5_md_t);
     int devx_srq_requested = (dc ?
                               (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_DC_SRQ) :
                               (md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_RC_SRQ));
-    return devx_srq_requested && (md->flags & UCT_IB_MLX5_MD_FLAG_RMP);
+    return devx_srq_requested &&
+           (UCT_RC_MLX5_TM_ENABLED(iface) ||
```

Comment:
does it mean we cannot use cyclic srq w/o HWTM enabled?

Reply:
no, we can use cyclic SRQ w/ non-HWTM if RMP is enabled

Reply:
the why we need to check for UCT_RC_MLX5_TM_ENABLED if we already check for RMP?
the user enabling HWTM is independent of having the HW support for cyclic SRQ

Reply:
we are checking whether HWTM is enabled or not for creating DevX XRQ
RMP check is needed for SWTM

Reply:
if user set UCT_IB_MLX5_MD_FLAG_DEVX_*C_SRQ, it means user wants DevX srq regardless of HWTM

Reply:
ok, then we should follow PRM and check for `HCA_CAP.log_max_xrq` (if `> 0` means that XRQ can be created) as we do for RMP in this PR. right?

Reply:
don't we already check log_max_xrq today somewhere?

Reply:
> don't we already check log_max_xrq today somewhere?

no, I don't see the check for this. we have `TODO` for this [here](https://github.com/openucx/ucx/blob/0dbd374b9c44c596e2c32d1ca372961e383af3fb/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L628)

Reply:
ok let's add then

Reply:
maybe rename the flag from UCT_IB_MLX5_MD_FLAG_MP_RQ to UCT_IB_MLX5_MD_FLAG_MP_XRQ, check  log_max_xrq in order to enable this flag, and in uct_rc_mlx5_is_devx_srq_supported() check for (TM_ENABLE && UCT_IB_MLX5_MD_FLAG_MP_XRQ)

Reply:
> maybe rename the flag from UCT_IB_MLX5_MD_FLAG_MP_RQ to UCT_IB_MLX5_MD_FLAG_MP_XRQ, check log_max_xrq in order to enable this flag

yes, it should work
but I think we have to check for `ext_stride_num_range` and set `UCT_IB_MLX5_MD_FLAG_MP_RQ` if it's supported in order to check for MP XRQ support
I'm going to add `UCT_IB_MLX5_MD_FLAG_XRQ` that will show whether XRQ supported or not by checking `log_max_xrq` capability in the same way as we do for RMP

Reply:
pls, take a look into my approach (the latest commit). I guess, it should be ok